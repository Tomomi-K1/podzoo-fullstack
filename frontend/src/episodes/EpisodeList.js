import React, { useState, useEffect } from "react";
import {useOutletContext} from 'react-router-dom'
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import EpisodeCard from './EpisodeCard';
/**show list of episode. nest EpisodeCard */

function EpisodeList(){
    const{count, episodes, link} =useOutletContext()
    console.debug('episodeList')
    console.log(count, episodes)
    return(
        // add hover effect
        <Box sx ={{pt:5}}>
            <Typography variant='body2' sx={{p:4}}>Number of Episodes shown here is {count}. For more episode, visit <a href={link} target="_blank">Podcast Homepage</a></Typography>
          
            <Grid container spacing={1} sx={{flexGrow:1}} justifyContent="center">
                <Grid item xs={12} md={11} lg={10} sx={{ display:"flex", justifyContent:"center"}}>
                    <Grid container justifyContent="center" spacing={1}>
                        {episodes.map(episode =>{
                            return (         
                                <Grid key={episode.id} item xs ={5} md={3} lg={2} sx={{display:"flex", justifyContent:"center"}}>
                                    <EpisodeCard episode={episode} />
                                </Grid>
                            
                            )})
                        }
                    </Grid>
                </Grid>
            </Grid>
            
        </Box>
        
    )
}

export default EpisodeList;