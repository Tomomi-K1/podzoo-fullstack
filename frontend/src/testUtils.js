import React from "react";
import UserContext from "./UserContext";

const demoUser = {
  username: "testuser",
  email: "test@test.net",
  fav:[1,2,3]
};

const favoritesArray =[1, 2]

const UserProvider =
    ({ children, currentUser = demoUser, checkFavPod=()=>true, favorites= favoritesArray}) => (
    <UserContext.Provider value={{ currentUser, checkFavPod, favorites }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };