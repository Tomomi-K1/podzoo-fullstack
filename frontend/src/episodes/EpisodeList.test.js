import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import * as rrd from 'react-router-dom';
import { UserProvider } from "../testUtils";
import EpisodeList from "./EpisodeList";
const mockData ={ 
    count: 100 ,
    episodes:[
        {artwork:"https://image.ausha.co/P0TYweU74SHBPcq5XiSwAEa1QzMUjsXfkX8YL2fq_1400x1400.jpeg?t=1616175875",
        author:"TheBBoost",
        categories:{9: 'Business'},
        description:"Comment développer son business ? Augmenter rapidement son chiffre d'affaires ? Trouver des  clients sans prospecter ? \"Le podcast J'peux pas j'ai business\" aborde les meilleures stratégies business, expliquées pas-à-pas et vous livre le plan d'action étape par étape pour construire le business et la vie de vos rêves.\nPour qui souhaite vivre de sa passion et découvrir la liberté financière, ce podcast creuse des sujets comme le marketing, l'entrepreneuriat, la prospection, l'état d'esprit, l'organisation et bien plus encore.\nÉnergie, stratégies, astuces et conseils d'une coach business certifiée sont au rendez-vous.\nPar Aline Bartoli, coach d'affaires chez TheBBoost.",
        feedId:75761,
        language:"fr",
        link:"http://www.thebboost.fr/",
        medium:"podcast",
        title:"J'peux Pas J'ai Business par TheBBoost",
        url:"https://feed.ausha.co/blDjes1L9aWL"}
        ],
    link:'testurl'
}

jest.mock('react-router-dom');

it('render with outlet context info', async() => {
    rrd.useOutletContext.mockReturnValue(mockData);
    render(<EpisodeList />) 
})

it('matches snapshot', async () => {
    rrd.useOutletContext.mockReturnValue(mockData);
    const { asFragment } = render(
        <MemoryRouter>
            <EpisodeList />
        </MemoryRouter>
    ) 
    expect(asFragment()).toMatchSnapshot();
})

it('render information from mockData', async () => {
    rrd.useOutletContext.mockReturnValue(mockData);
    render(
        <MemoryRouter>
            <EpisodeList />
        </MemoryRouter>
    ) 
    expect(screen.getByText("J'peux Pas J'ai Business par TheBBoost")).toBeInTheDocument();
})
