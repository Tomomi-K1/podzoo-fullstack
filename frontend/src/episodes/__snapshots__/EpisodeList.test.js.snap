// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-pxpt32"
  >
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-8gubce-MuiTypography-root"
      >
        Number of Episodes shown here is 100. For more episode and details, visit
        <a
          href="testurl"
          target="_blank"
        >
           Podcast Homepage
        </a>
      </p>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-td5piu-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-9 MuiGrid-grid-lg-7 css-svdq89-MuiGrid-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root undefined css-15xjbqe-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiBox-root css-1nqrw45"
              >
                <img
                  alt="episode artwork"
                  class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-gganah-MuiCardMedia-root"
                  src="default.jpg"
                />
              </div>
              <div
                class="MuiBox-root css-1ylu0bo"
              >
                <div
                  class="MuiCardContent-root css-1k2e1n7-MuiCardContent-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1nebxi1-MuiTypography-root"
                  >
                    J'peux Pas J'ai Business par TheBBoost
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1yeu3gx-MuiTypography-root"
                  >
                    Published: 
                  </p>
                  <audio
                    controls=""
                    type=""
                  />
                </div>
                <button
                  aria-expanded="false"
                  aria-label="show more"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-3ogxn6-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ExpandMoreIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </li>
        </div>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
